### ISSUE TITLE
-l option requires sudo

### ISSUE THREAD

**Issue by jwbensley** [jwbensley/Etherate#39]
##### ISSUE TYPE

- [X] Bug Report
- [ ] Enhancement/Feature Request
- [ ] Question

##### VERSION DETAILS
```
$ ./etherate -V
Etherate version 1.15 2017-10
```

##### SUMMARY
Using the `-l` option exits with an error because sudo is required. Can this be prevented?

##### STEPS TO REPRODUCE
```
$ ./etherate -l
Couldn't close socket : Bad file descriptor
```

**Comment by jwbensley** [comment]
No it can't be prevented, `ioctl(SIOCGIFHWADDR)` requires UID 0. Will move the check uid code to be before the CLI args are parsed.

**Comment by jwbensley** [comment]
This is fixed in version 1.17 (2018-04).
